### PR TITLE
feat: check filter against lightPush, basic implementation

### PR DESCRIPTION
**Pull request by weboko**
## Problem

We observe some full nodes in the network that accept `subscription` request from `js-waku` but do not push messages later.

## Solution

As mitigation for it implement basic handler for sending ephemeral message during subscription setup and check.

## Notes

- Related to https://github.com/waku-org/js-waku/issues/2158

### CODE REVIEW

File: packages/message-hash/src/index.ts [hunk 1]
```diff
@@ -56,6 +56,6 @@ export function messageHashStr(
   message: IProtoMessage | IDecodedMessage
 ): string {
   const hash = messageHash(pubsubTopic, message);
-  const hashStr = bytesToUtf8(hash);
+  const hashStr = bytesToHex(hash);
```
Comment (weboko) [inline packages/message-hash/src/index.ts:59]: this is fix for `messageHashStr` when it doesn't transform bytes properly and shows weird symbols

**Comment by weboko** [comment]
This is temporary approach and should be revisited in https://github.com/waku-org/js-waku/issues/2075

File: packages/interfaces/src/filter.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export type SubscriptionCallback<T extends IDecodedMessage> = {
 export type SubscribeOptions = {
   keepAlive?: number;
   pingsBeforePeerRenewed?: number;
-  maxMissedMessagesThreshold?: number;
+  enableLightPushFilterCheck?: boolean;
```
Comment (adklempner) [inline packages/interfaces/src/filter.ts:23]: note that removing `maxMissedMessagesThreshold` is a breaking change

File: packages/sdk/src/protocols/filter/subscription_manager.ts [hunk 3]
```diff
@@ -109,6 +135,11 @@ export class SubscriptionManager implements ISubscription {
         callback
       } as unknown as SubscriptionCallback<IDecodedMessage>;
 
+      // don't handle case of internal content topic
+      if (contentTopic === this.buildLightPushContentTopic()) {
```
Comment (adklempner) [inline packages/sdk/src/protocols/filter/subscription_manager.ts:139]: should this be placed earlier in the code?

File: packages/interfaces/src/filter.ts (hunk 2 above)
Comment (weboko) [reply packages/interfaces/src/filter.ts:23]: it is but it wasn't used afaik

File: packages/sdk/src/protocols/filter/subscription_manager.ts (hunk 3 above)
Comment (weboko) [reply packages/sdk/src/protocols/filter/subscription_manager.ts:139]: this should be in this callback unfortunately

File: packages/interfaces/src/filter.ts (hunk 2 above)
Comment (adklempner) [reply packages/interfaces/src/filter.ts:23]: if it's exposed externally we don't really know if consumers use it. But it's probably not a huge fix on their end if that's the case

**Review by adklempner — APPROVED** [review]

**Comment by weboko** [comment]
> I notice browser tests are failing now, unrelated to the PR though

I suppose there was an outage and wrong versions / configs were used lol

```
15 10 2024 23:15:55.796:ERROR [launcher]: ChromeHeadless stderr: [7983:7983:1015/231555.287086:FATAL:zygote_host_impl_linux.cc(128)] No usable sandbox! If you are running on Ubuntu 23.10+ or another Linux distro that has disabled unprivileged user namespaces with AppArmor, see https://chromium.googlesource.com/chromium/src/+/main/docs/security/apparmor-userns-restrictions.md. Otherwise see https://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md for more information on developing with the (
```

**Pull request merged**